### PR TITLE
remove some redundant awaits

### PR DESCRIPTION
**Pull request by SimonCropp**
reboot of #11644

@ajcvickers  i think this one includes all your feedback

### CODE REVIEW

File: src/EFCore/DbContext.cs [hunk 1]
```diff
@@ -726,27 +726,24 @@ private void SetEntityState(InternalEntityEntry entry, EntityState entityState)
...
-        private async Task SetEntityStateAsync(
+        private Task SetEntityStateAsync(
             InternalEntityEntry entry,
             EntityState entityState,
             CancellationToken cancellationToken)
         {
             if (entry.EntityState == EntityState.Detached)
             {
-                await DbContextDependencies.EntityGraphAttacher.AttachGraphAsync(
+                return DbContextDependencies.EntityGraphAttacher.AttachGraphAsync(
```
Comment (roji) [inline src/EFCore/DbContext.cs]: Change to expression-bodied method while we're at it?

Comment (roji) [inline src/EFCore/DbContext.cs]: I may be misreading the code, but isn't the following possible?

```suggestion
        private Task SetEntityStateAsync(
            InternalEntityEntry entry,
            EntityState entityState,
            CancellationToken cancellationToken)
        => entry.EntityState == EntityState.Detached
            ? DbContextDependencies.EntityGraphAttacher.AttachGraphAsync(
                entry,
                entityState,
                forceStateWhenUnknownKey: true,
                cancellationToken: cancellationToken)
            : entry.SetEntityStateAsync(
                entityState,
                acceptChanges: true,
                forceStateWhenUnknownKey: entityState,
                cancellationToken: cancellationToken);
                cancellationToken: cancellationToken);
```

Comment (roji) [reply src/EFCore/DbContext.cs]: I may be misreading the code, but isn't the following possible?

```suggestion
        private Task SetEntityStateAsync(
            InternalEntityEntry entry,
            EntityState entityState,
            CancellationToken cancellationToken)
        => entry.EntityState == EntityState.Detached
            ? DbContextDependencies.EntityGraphAttacher.AttachGraphAsync(
                entry,
                entityState,
                forceStateWhenUnknownKey: true,
                cancellationToken: cancellationToken)
            : entry.SetEntityStateAsync(
                entityState,
                acceptChanges: true,
                forceStateWhenUnknownKey: entityState,
                cancellationToken: cancellationToken);
                cancellationToken: cancellationToken);
```

Comment (SimonCropp) [reply src/EFCore/DbContext.cs]: but doesnt that make it less readable?

Comment (roji) [reply src/EFCore/DbContext.cs]: I personally always use expressions where possible, it's definitely more readable to me. But maybe someone else can with in.

Comment (SimonCropp) [reply src/EFCore/DbContext.cs]: The expression is fine. It is the ternary that i think makes it more difficult to grok. 

But either way. Isnt that out of scope for this change?

**Comment by SimonCropp** [comment]
gentle bump

File: src/EFCore/DbContext.cs (hunk 1 above)
Comment (ajcvickers) [reply src/EFCore/DbContext.cs]: @SimonCropp @roji Did we come to any conclusion on this?

Comment (SimonCropp) [reply src/EFCore/DbContext.cs]: i gave up and moved on. sorry.

**Comment by roji** [comment]
I'd like us to reopen this and merge - removing redundant awaits is definitely a good idea (for perf). We were only blocking on a minor question of style, which frankly I don't care about that much.

@SimonCropp sorry this stagnated for a while, these are busy times... @ajcvickers can we please reopen and merge if you're OK with it?

**Comment by ajcvickers** [comment]
@roji Sure, go for it, but since the repository was deleted I'm not sure what git/github magic is needed.

**Comment by roji** [comment]
@SimonCropp would you mind re-submitting this please?

**Comment by SimonCropp** [comment]
@roji given how small the changes were. isnt it something u can re-apply relatively simply?

**Comment by roji** [comment]
@SimonCropp I certainly can, but then you wouldn't be credited for it, and the commit seems to be gone since you deleted the repo (I hope that doesn't mean you won't be contributing anymore!).

**Comment by SimonCropp** [comment]
No credit needed. I was just trying to give back a little, since i dont pay for EF. 

Re me deleting the repo: i like to regularly clean up my forks. If it doesnt look like a pr will be accepted I delete the fork. 

Yes i will be contributing in the future.

**Comment by roji** [comment]
Submitted #14724

**Pull request closed without merging**